Adds a note that source maps are not uploaded for debug mode in React Native / Expo (#11478)
getsentry/sentry-docs: docs/platforms/react-native/sourcemaps/uploading/expo.mdx

## Automatic Upload for Release

> When the Sentry Expo Plugin `@sentry/react-native/expo` and the Sentry Metro Pugin are added to your application configuration, source maps for the native builds are uploaded automatically during EAS Builds and when building the native application release locally using `npx expo prebuild`. Note that uploading source maps for debug builds is disabled, as symbolication is handled by the Metro dev server.

Reviewer: ```suggestion
When the Sentry Expo Plugin `@sentry/react-native/expo` and the Sentry Metro Plugin are added to your application configuration, source maps for the native builds are uploaded automatically during EAS Builds and when building the native application release locally using `npx expo prebuild`. Because symbolication is already handled by the Metro dev server, automatic uploading of source maps for debug builds is disabled.
```

> When the Sentry Expo Plugin `@sentry/react-native/expo` and the Sentry Metro Pugin are added to your application configuration, source maps for the native builds are uploaded automatically during EAS Builds and when building the native application release locally using `npx expo prebuild`. Note that uploading source maps for debug or simulator builds is disabled, as symbolication is handled by the Metro dev server.

Reviewer 2: ```suggestion
When the Sentry Expo Plugin `@sentry/react-native/expo` and the Sentry Metro Pugin are added to your application configuration, source maps for the native builds are uploaded automatically during EAS Builds and when building the native application release locally using `npx expo prebuild`. Note that uploading source maps for debug builds is disabled, as symbolication is handled by the Metro dev server.
```

- `dist` is the default output directory of `eas update`.
- Uploaded source maps have no associated releases. This is expected as updates can apply to multiple releases.

> - Uploading of source maps for debug or simulator builds is disabled.

Reviewer 2: These are notes for EAS Updates, meaning for situation when a user creates a JS bundle to distribute as ota update. 

This note is relevant for the previous section `[Automatic Upload for Release](https://docs.sentry.io/platforms/react-native/sourcemaps/uploading/expo/#automatic-upload-for-release)`

Author: Good catch @krystofwoldrich 🙇  I've moved the note to the correct section adding a short sentence for the why with 6d154f79f78b7773d441d927d35ff2e81e46ebb8.

> I think we should include the information also on the main source maps page. 

Also added a separate a bit more extended section on the main source maps page with e339fd8a050c0d61445cd692164855eb0591491b

Reviewer 2: Thank you. I suggest removing the sim builds as release for sim will upload the source maps.

I also added @lizokm to review the docs style.

Author: Thank you @krystofwoldrich! Makes sense. I've committed your suggestions 🙇
